hipparchus-core/src/main/java/org/hipparchus/fraction/ConvergentsIterator.java

```diff
@@ -21,112 +21,97 @@
+        private final long   p1;
+
+        /** Denominator of current convergent. */
+        private final long   q1;
+
+        /** Remainder of current convergent. */
+        private final double r1;
+
+        private ConvergenceStep(final long p0, final long q0, final long p1, final long q1, final double r1) {
+            this.p0 = p1;
+            this.q0 = q1;
+            final long a1 = (long) FastMath.floor(r1);
+            try {
+                this.p1 = FastMath.addExact(Math.multiplyExact(a1, p1), p0);
+                this.q1 = FastMath.addExact(Math.multiplyExact(a1, q1), q0);
+                this.r1 = 1.0 / (r1 - a1);
+            } catch (ArithmeticException e) { // unlike the name implies FastMath's multiplyExact() is slower
+                throw new MathIllegalStateException(LocalizedCoreFormats.FRACTION_CONVERSION_OVERFLOW, r1, p1, q1);
+            }
+        }
 
-    /** Count of already generated convergents. */
-    private int     n;
+        public static ConvergenceStep start(double value) {
+            return new ConvergenceStep(0, 1, 1, 0, value);
+        }
 
-    /** Simple constructor.
-     * @param value value towards which convergents should converge
-     * @param maxConvergents maximum number of convergents to generate
-     * @param builder builder for the convergents
-     */
-    ConvergentsIterator(final double value, final int maxConvergents, final Builder<T> builder) {
+        public ConvergenceStep next() {
+            return new ConvergenceStep(p0, q0, p1, q1, r1);
+        }
 
-        this.value          = value;
-        this.maxConvergents = maxConvergents;
-        this.builder        = builder;
+        public long getNumerator() {
+            return p1;
+        }
 
-        // initialize iterations
-        p0   = 0;
-        q0   = 1;
-        p1   = 1;
-        q1   = 0;
-        r1   = value;
-        stop = false;
-        n    = 0;
+        public long getDenominator() {
+            return q1;
+        }
 
-    }
+        public double getFractionValue() {
+            return getNumerator() / (double) getDenominator();
+        }
 
-    /** {@inheritDoc} */
-    @Override
-    public boolean hasNext() {
-        return n < maxConvergents && !stop;
+        @Override
+        public String toString() {
+            return getNumerator() + "/" + getDenominator();
+        }
     }
 
-    /** {@inheritDoc} */
-    @Override
-    public T next() {
-        ++n;
-
-        final long a1 = (long) FastMath.floor(r1);
-        long p2 = (a1 * p1) + p0;
-        long q2 = (a1 * q1) + q0;
-
-        final double convergent = (double) p2 / (double) q2;
-        stop = Precision.equals(convergent, value, 1);
-        if ((p2 > OVERFLOW || q2 > OVERFLOW) && !stop) {
-            throw new MathIllegalStateException(LocalizedCoreFormats.FRACTION_CONVERSION_OVERFLOW, value, p2, q2);
+    static Pair<ConvergenceStep, Boolean> convergent(double value, int maxIterations,
```

Comment:
Do you think this method should just return a ConvergenceStep without the flag that indicated if the value has converged?
I added it with the intention that users could only want a value if it has converged but at least the direct usage in (Big-)Fraction most of the time requires just the last value regardless if hat has passed the convergence test.

If just the step is returned it should be stated in the doc (I still have to write) that the returned fraction is not guaranteed to fulfill the convergence-test and the convergence-test is more a criteria to abort the iteration prematurely.

Reply:
No, your design is fine to me.

So I just wait for the doc and will merge the PR.

Thanks again for your contribution!

Reply:
> No, your design is fine to me.
> 
> So I just wait for the doc and will merge the PR.

Done.
> 
> Thanks again for your contribution!

You are very welcome and thanks for Hipparchus!